core/market/src/negotiation/provider.rs

```diff
@@ -139,24 +147,26 @@ impl ProviderBroker {
     // TODO: this is only mock implementation not to throw 501
     pub async fn reject_proposal(
         &self,
-        _offer_id: &SubscriptionId,
+        offer_id: &SubscriptionId,
         proposal_id: &ProposalId,
         id: &Identity,
+        reason: Option<Reason>,
     ) -> Result<(), ProposalError> {
-        // self.common
-        //     .reject_proposal(offer_id, proposal_id, OwnerType::Provider)
-        //     .await?;
-        //
+        self.common
+            .reject_proposal(offer_id, proposal_id, OwnerType::Provider, &reason)
+            .await?;
+
         // self.api
         //     .reject_proposal(id, proposal_id, TODO: requestor_id)
         //     .await
         //     .map_err(|e| ProposalError::Send(proposal_id.clone(), e))?;
 
         counter!("market.proposals.provider.rejected", 1);
         log::info!(
-            "Provider {} rejected Proposal [{}]",
+            "Provider {} rejected Proposal [{}] with reason: {:?}",
```

Comment:
Please don't debug. I implemented display for this.

Reply:
o, I was not aware. Where is it?

Reply:
https://github.com/golemfactory/ya-client/blob/bcc8db5ba7d628c56c98c0213c9ee57069706941/model/src/market/reason.rs#L7
On branch `flatten-json-in-JsonReason`